packages/storage/src/components/ImageGallery/image.js

```diff
@@ -1,19 +1,25 @@
-import React, { useState, useEffect } from 'react';
+import React, { useState, useEffect, useRef } from 'react';
 import PropTypes from 'prop-types';
-import { Spinner } from 'react-rainbow-components';
+import { Spinner, ButtonIcon } from 'react-rainbow-components';
+import { Trash, Check } from '@rainbow-modules/icons';
+import { confirmModal } from '@rainbow-modules/app';
 import RenderIf from 'react-rainbow-components/components/RenderIf';
 import TruncatedText from 'react-rainbow-components/components/Structural/truncatedText';
 import {
     StyledFileContainer,
     StyledContainerImage,
     StyledContainerSpinner,
     StyledImage,
+    StyledDeleteButton,
+    StyledCheckContainer,
+    TrashIcon,
 } from './styled';
 
 export default function Image(props) {
-    const { imageRef, onSelect, onError } = props;
+    const { imageRef, onSelect, onError, allowDelete, handleDeleteClick } = props;
```

Comment:
when passing a callback function as a prop we use to start its name with on... like onDelete, the function that handles the callback is the one that is called like this with handle...